src/buildApplication/generateGlobalJs.ts

```diff
@@ -8,7 +8,25 @@ import { cliWarn } from '../cli';
  */
 export function generateGlobalJs(): string {
 	const nodeEnv = getNodeEnv();
-	return `globalThis.process = { env: { NODE_ENV: '${nodeEnv}' } };`;
+
+	return `
+		import { AsyncLocalStorage } from 'node:async_hooks';
+		globalThis.AsyncLocalStorage = AsyncLocalStorage;
+
+		const __ENV_ALS__ = new AsyncLocalStorage();
+		__ENV_ALS__['NODE_ENV'] = '${nodeEnv}';
```

Comment:
Is that definitely correct? I'm still getting familiar with ALS APIs, but it seems a bit weird to be setting a value directly on the ALS instance here, rather than on the store.

Does `__ENV_ALS__.getStore()` return something that references back to props assigned on the `__ENV_ALS__` instance somehow?

Double-check that `process.env.NODE_ENV` and `process.env.SOME_OTHER_SET_ENV_VAR` are set properly in an example project.

Reply:
nono, to be totally clear, `__ENV_ALS__['NODE_ENV']` is a hack, I just wanted to have the `NODE_ENV` available and didn't want to create a new variable for it, so I am just piggybacking it on top of the ASL (as a plain object)

(
you can see here: https://github.com/cloudflare/next-on-pages/blob/cbc3395270816c20a4f60f980e6679e11cc4788b/templates/_worker.js/index.ts#L15 that as soon as I can, I take it and put it in the actual store
)

Alternatively I can have a `__NODE_ENV__` variable instead (I don't think I can use it in the ASL initialization)

Reply:
PS: I did check that I get all the `process.env` variables in an example project 😅 (specifically here: https://github.com/dario-piotrowicz/next-apps-for-testing/blob/802e9478a3b322f9b2850008aec9e4abd7bc8215/apps/simple-app-dir-13.3.1/app/api/hello/route.js)

Reply:
ohhh... yeah, I am not a fan of that, I'm afraid. Why not just do `__ENV_ALS__.run({ ...env, NODE_ENV: ${nodeEnv} })`?

Reply:
Ah, because it's in a separate unprocessed file. I'd maybe use esbuild's `define` feature then.

`__ENV_ALS__.run({ ...env, NODE_ENV: __NODE_ENV__ })`

and `define({ __NODE_ENV__: "production" })` in the esbuild command.

Reply:
Ah, in fact you've changed it even more. I'm just now seeing `getNodeEnv()` too.

Okay, whatever. Last suggestion would be to put it in the proxy maybe?

```
get (_, property) => {
  if (property === 'NODE_ENV') {
    return ${nodeEnv}
  } else {
    return Reflect.get(__ENV_ALS__.getStore(), property)
  }
}
```

But not the end of the world if you ship what you have right now.

Reply:
test...

Reply:
done! (https://github.com/cloudflare/next-on-pages/pull/216/commits/97b1afb1501ca5c1dbe6692335c9b9abd4859926)

Thank you so very much for the suggestion! 🙏  (and sorry for the silly mistake 😓)